### DISCUSSION TITLE
Welcome to XQuartz Discussions!

### DISCUSSION THREAD

**Discussion by jeremyhu** [XQuartz/XQuartz#23] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by 21tesla** [comment]
**Thank you for your dedication and hard work on Xquartz.** In my field of structural biology, we rely on X11 software to do our research. 

We still use Sun **libxview**, as well - a graphics library released in 1989. The last time there was a successful build of it was 2005 for ppc and i386. When MacOS switched exclusively to x86_64 (and let's not even talk about arm64), we lost a critical data processing program. 

I retrieved the old xview code from my archive dated 2010 (probably the last attempt). I have since made some changes and placed it all in **https://github.com/21tesla/libxview-macos** with a simple Makefile in the src/ directory. Of the 487 source code files, all but 3 compile and I am stuck. One file needs the help of someone who can apply the correct X11 call. The other two files are more general. 

I post this here in the hope someone with X11 experience would be willing to help.

Thank you
logan donaldson
logand@yorku.ca

**Comment by terry9873** [comment]
Hi, I am desperately hoping someone can help me. I am fairly new to Linux, at least from a CLI perspective. 
I have just set up my first Linux VPS, and I use a Mac here, so I am trying to connect to the VPS from Mac. 
I have SSH access, set up keypairs for increase security, and I have gone through the process of setting up X2GO according to some youtube videos I watched and followed. Thing is.... it doesn't work, yet I did everything everyone else did in the videos. 

I keep getting the error "X2GO Client could not start X11 server". 

I read the instructions, I installed XQuartz.app which is sat in my Utilities folder and is running. Yet I still get this error when trying to make my first connection to the VPS. I would be very grateful for any help. 
Thanks

**Reply by jeremyhu** [reply]
I'm not familiar with X2Go, but perhaps you didn't logout and back in as required with the first install.

**Comment by muesic** [comment]
G'day Jeremy - wow, glad you're back! As I recall, you were the very soul of XQuartz way back when it was still active (as was Ben Byer before you). I've been getting by using XQuartz (actually using MacPorts currently) and I've been suffering every day with one particularly nasty super-slow graphics bug for a few years now. It's covered in the following link:
https://gitlab.freedesktop.org/xorg/xserver/-/issues/188
Is there any chance a fix for this is in XQuartz's future?

**Reply by jeremyhu** [reply]
That is, unfortunately, a bug in macOS and not in XQuartz.  It would require redesigning quite a bit of our implementation to get it working right.  That effort would be better focused on redesigning XQuartz as a compositing window manager... or maybe even making something like MacWayland and running XWayland as the X11 server (ie: X11 client <-> XWayland <-> MacWayland <-> AppKit).

**Reply by muesic** [reply]
Thanks for your response. Bummer! Given that it's a bug in macOS I'm guessing there will be zero interest in fixing it from Apple because we pesky X11 folks simply don't rate. Remember the good old days when Apple actually supported the sciences, sigh...
I'm also guessing your other options aren't likely to show up in my working lifetime (given I'm so old), double sigh...
Sorry for being such a wet blanket. Please don't let that trick you into thinking I don't appreciate that you're back on board - that's the big silver lining to all this :-)

**Comment by HeyMerlin** [comment]
Just noticed that XQuartz development is back (thanks to an upgrade notice when I launched it)!  

@jeremyhu a big thank you for your efforts. If you have a way for folk to buy you a coffee (or beverage of choice), post it please.

I've been using XQuartz for years and now we have several students that use it regularly.

Cheers.

**Reply by jeremyhu** [reply]
Sorry it took so long, and sorry to those of you who didn't intend to get beta updates.  I had no idea 2.7.11 had the incorrect feed URL in it =(.  At least I didn't push the early alpha releases to the SU channel =).

**Comment by tehcog** [comment]
@jeremyhu thanks for making XQuartz work with Mavericks.  So far, so good here with XQuartz 2.8.0 (xorg-server 1.19.7)  :)

**Comment by ScattyNatalie** [comment]
I know that this may seem like a interesting topic. But is there any chance of porting this to iOS? We have Xorg working but nothing to display it with. 
![5315C6C5-AEEF-4290-BA4D-B23D7C21A2B9](https://user-images.githubusercontent.com/55960119/115052753-4f7ba980-9eac-11eb-94e2-9f98097480b0.png)
Any input into this option would be greatly appreciated. It’s also ok for you to say no to this idea, I won’t mind.

**Reply by dazuelos** [reply]
How do you get this nice keyboard on `iOS`?

**Comment by hikerguy1900** [comment]
@jeremyhu thank you for your work on XQuartz. I use it every day.

**Comment by MorelloCherry** [comment]
Hi,

could you please confirm that XQuartz uses hardware acceleration?

I am running XQuartz on a 2013 MacPro which has been upgraded to MacOS 11.4 and when running a 4K resolution at 60HZ in 4:2:0 I can see individual lines being rendered in gvim. If I switch to 4:2:2 then everything seems sped up but not what I saw on older MacOS releases.

Thanks a lot

**Reply by jeremyhu** [reply]
Rendering in XQuartz is naive and very very sub-optimal.

**Reply by MorelloCherry** [reply]
Given that Apple has removes X11 from MacOS years ago I can imagine that improving rendering is a major task. Out of curiosity, is there someone looking at this?

**Reply by jeremyhu** [reply]
no.  there is nobody actively working on this.
